src/manifolds/ProductManifold.jl

```diff
@@ -19,6 +19,7 @@ struct ProductManifold{TM<:Tuple} <: Manifold
 end
 
 ProductManifold(manifolds::Manifold...) = ProductManifold{typeof(manifolds)}(manifolds)
```

Comment:
Just to be safe, it might be good to check that the `manifolds`-Tuple as a (strictly) positive length, since that would also lead to an zero-product product manifold

Reply:
How would such a tuple be created? If one passes no args to `ProductManifold`, the below constructor is called, which throws the error. Only if one passes at least one arg, creating a strictly positive length tuple, would this constructor be used.

Reply:
Ah, I did not try that, but you're right, that might not be possible, since Tuples are immutable, for Arrays this is different, for sure.